### PR TITLE
[ci] add `mypy` to as a part of `qa`

### PR DESCRIPTION
**Pull request by nowei**
Changes:
* Adds `mypy` step to QA checks

TO BE MERGED AFTER: https://github.com/usdigitalresponse/cpf-reporter/pull/506

### CODE REVIEW

File: .github/workflows/qa.yml [hunk 1]
```diff
@@ -286,6 +288,35 @@ jobs:
...
+          ref: ${{ inputs.ref }}
+          show-progress: 'false'
+          persist-credentials: 'false'
+      - uses: actions/setup-python@v3
+        with:
+          python-version: '3.12'
+      - run: pip install mypy
+      - name: Get Python changed files
+        id: changed-py-files
+        uses: tj-actions/changed-files@v23
```
[the remark this answers is not in the document]
Comment (nowei) [reply .github/workflows/qa.yml]: Addressed.

File: .github/workflows/qa.yml [hunk 2]
```diff
@@ -286,6 +288,35 @@ jobs:
...
+      - uses: step-security/harden-runner@446798f8213ac2e75931c1b0769676d927801858 # v2.10.0
+        with:
+          disable-sudo: true
+          egress-policy: audit
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          ref: ${{ inputs.ref }}
+          show-progress: 'false'
+          persist-credentials: 'false'
+      - uses: actions/setup-python@v3
```
[the remark this answers is not in the document]
Comment (nowei) [reply .github/workflows/qa.yml]: Addressed.

File: .github/workflows/qa.yml [hunk 3]
```diff
@@ -286,6 +288,35 @@ jobs:
...
+        with:
+          python-version: '3.12'
```
[the remark this answers is not in the document]
Comment (nowei) [reply .github/workflows/qa.yml]: Ahh, that sounds better, I'll do that

File: .github/workflows/qa.yml [hunk 4]
```diff
@@ -286,6 +288,35 @@ jobs:
...
+      - run: pip install mypy
```
[the remark this answers is not in the document]
Comment (nowei) [reply .github/workflows/qa.yml]: Added it to the dev dependencies

File: .github/workflows/qa.yml [hunk 5]
```diff
@@ -286,6 +288,35 @@ jobs:
...
+      - name: Get Python changed files
+        id: changed-py-files
+        uses: tj-actions/changed-files@v23
+        with:
+          files: |
+            *.py
+            **/*.py
+      - name: Run if any of the listed files above is changed
+        if: steps.changed-py-files.outputs.any_changed == 'true'
+        run: mypy ${{ steps.changed-py-files.outputs.all_changed_files }} --ignore-missing-imports
```
[the remark this answers is not in the document]
Comment (nowei) [reply .github/workflows/qa.yml]: Done

**Comment by nowei** [comment]
Thanks for the feedback! I believe I addressed all the comments, but let me know if I missed anything 😄

**Review by TylerHendrickson — APPROVED** [review]
@nowei Looks great – approved!

**Pull request merged**